File: 1752-arithmetic-subarrays/arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {

        int n = nums.length;
        int m = l.length;
        List<Boolean> res = new ArrayList<>();

        for (int i = 0; i < m; i++) {
            int a = l[i];
            int b = r[i];

            int[] arr = Arrays.copyOfRange(nums, a, b + 1);

            Arrays.sort(arr);
            int diff = arr[1] - arr[0];
            int j = 2;

            while (j < arr.length) {
                if (arr[j] - arr[j - 1] != diff) {
                    res.add(false);
                    break;
                }
                j++;
            }

            if (j == arr.length) {
                res.add(true);
            }
        }

        return res;
    }
        
    
}

File: 1752-arithmetic-subarrays/arithmetic-subarrays.js
/**
 * @param {number[]} nums
 * @param {number[]} l
 * @param {number[]} r
 * @return {boolean[]}
 */
var checkArithmeticSubarrays = function(nums, l, r) {
     const n = nums.length;
        const m = l.length;
        const res = [];

        for (let i = 0; i < m; i++) {
            const a = l[i];
            const b = r[i];

            const arr = nums.slice(a, b + 1);

            arr.sort((x, y) => x - y);
            const diff = arr[1] - arr[0];
            let j = 2;

            while (j < arr.length) {
                if (arr[j] - arr[j - 1] !== diff) {
                    res.push(false);
                    break;
                }
                j++;
            }

            if (j === arr.length) {
                res.push(true);
            }
        }

        return res;
};


